buildpacks/frontend-selector/bin/detect

```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env bash
+set -eo pipefail
+
+# 1. GET ARGS
+plan_path=$2
+
+cat >>"${plan_path}" <<EOL
+[[provides]]
+name = "frontend"
+
+[[requires]]
+name = "frontend"
+
+[requires.metadata]
+launch = true
+EOL
+
+# 2. DECLARE DEPENDENCIES (OPTIONAL)
+if [[ "${BP_REQUIRES}" == *"jupyterlab"* ]]; then
+cat >>"${plan_path}" <<EOL
+[[requires]]
+name = "jupyterlab"
+
+[requires.metadata]
+launch = true
```

Comment:
This is just a question for my understanding. This means that it will use the jupyterlab buildpack as a dependency to this buildpack? How does it know how to resolve the name to a path or uri or version? And futher below you just have "requires - vscodium" which is even weirder because vscodium is not a buildpack in this repo. Is this coming from the builder you have defined?

Reply:
Yes its from builder